src/frontend/js/widgets/Dashboard/utils/routers.tsx

```diff
@@ -113,14 +118,16 @@ const getDashboardRouteAttribute =
   (attribute: RouteAttributes) =>
   (intl: IntlShape) =>
   (
-    path: DashboardPaths,
+    path: DashboardPaths | TeacherDashboardPaths,
```

Comment:
IMO: If we want to split those pathes, we must keep a strict logic and rename `DashboardPaths` to `LearnerDashboardPaths` and something like this.

Furthermore we could imagine a type `DashboardPaths` which would be the union of both previous ones.

Reply:
I started by spliting the code in 3 parts: 
* `Dashboard`
* `LearnerDashboard`
* `TeacherDashboard`

It create lot's of noise during my development so i start fresh by only adding new `TeacherDashboard` stuff plugged to existing  `Dashboard`.
I'm not against doing this `Dashboard` / `LearnerDashboard` split but may be in a second time / PR.

> a type DashboardPaths which would be the union of both previous ones
I had that but type union didn't work the same as enums. We would not be able to use  `DashboardPathsUnionType[MY_ROUTE_ID]`.
I didn't find nice way to merge two enums, i think it's not possible.